test/unit/plugins/validate-semantic/2and3/schemas.js

```diff
@@ -169,3 +169,347 @@ describe(`validation plugin - semantic - 2and3 schemas`, () => {
+      })
+    })
+  })
+  
+  it("should not return an error when all items are number in a enum number type in OpenApi 3", () => {
+    const spec = {
+      openapi: "3.0.0",
+      "paths": {
+        "/pets": {
+          "get": {
+            "parameters": [
+              {
+                name: "number",
+                in: "query",
+                schema: {
+                  type: "number",
+                  enum: [1, 2, 3]
+                }
+              },
+            ]
+          }
+        }
+      }
+    }
+
+    return validateHelper(spec)
+    .then(system => {
+      const allErrors = system.errSelectors.allErrors().toJS()
+      expect(allErrors.length).toEqual(0)
+    })
+  })
+
+  //Array Tests
+
+  it("should return an error for a non array value in a enum array type in OpenApi 3", () => {
+    const spec = {
+      openapi: "3.0.0",
+      "paths": {
+        "/pets": {
+          "get": {
+            "parameters": [
+              {
+                name: "arraySample",
+                in: "query",
+                schema: {
+                  type: "array",
+                  enum: [1, 2, 3],     
+                  items: {
+                    type: "array",
+                    items: {
+                      type: "number"
+                    }             
+                  }
+                }
+              },
+            ]
+          }
+        }
+      }
+    }
+
+    return validateHelper(spec)
+    .then(system => {
+      const allErrors = system.errSelectors.allErrors().toJS().filter((err => err.source !== "")) 
+      expect(allErrors.length).toEqual(3)
+      expect(allErrors[0]).toInclude({
+        level: "warning",
+        message: "enum value should conform to its schema's `type`",
+        path: ["paths", "/pets", "get", "parameters", "0", "schema", "enum", 0]
+      })
+      expect(allErrors[1]).toInclude({
+        level: "warning",
+        message: "enum value should conform to its schema's `type`",
+        path: ["paths", "/pets", "get", "parameters", "0", "schema", "enum", 1]
+      })
+      expect(allErrors[2]).toInclude({
+        level: "warning",
+        message: "enum value should conform to its schema's `type`",
+        path: ["paths", "/pets", "get", "parameters", "0", "schema", "enum", 2]
+      })
+    })
+  })
+  
+  it("should not return a type error when all items are array in a enum array type in OpenApi 3", () => {
+    const spec = {
+      openapi: "3.0.0",
+      "paths": {
+        "/pets": {
+          "get": {
+            "parameters": [
+              {
+                name: "arraySample",
+                in: "query",
+                schema: {
+                  type: "array",
+                  enum: [[1,2],[3,4]],     
+                  items: {
+                    type: "number",
+                    items: {
+                      type: "number"
+                    }                         
```

Comment:
**Question**: `type: number` schemas can't have `items` — what is this intended to do?